### PR TITLE
feat(history): add history/requests endpoint

### PR DESCRIPTION
**Pull request by Tomas2D**
Ref: #102

### CODE REVIEW

File: src/client-types.ts [hunk 1]
```diff
@@ -148,3 +148,23 @@ export type PromptTemplateExecuteInput = z.input<
...
+// HISTORY
+
+export const HistoryInputSchema = ApiTypes.HistoryInputSchema.pick({
+  status: true,
+  origin: true,
+})
+  .extend({
+    count: z.number().int().min(1).nullish(),
+    offset: z.number().int().min(0).nullish(),
+  })
```
Comment (pilartomas) [inline src/client-types.ts:161]: `intersection` or `and`

File: src/client-types.ts [hunk 2]
```diff
@@ -148,3 +148,23 @@ export type PromptTemplateExecuteInput = z.input<
...
+  .partial();
+export type HistoryInput = z.input<typeof HistoryInputSchema>;
+
+export type HistoryOptions = HttpHandlerOptions;
+
+export const HistoryOutputSchema = ApiTypes.HistoryOutputSchema;
+export type HistoryOutput = z.output<
+  typeof HistoryOutputSchema
+>['results'][number];
```
Comment (pilartomas) [inline src/client-types.ts:170]: Not like this, the type MUST correspond to the schema. Use `.shape.results.element`

File: src/tests/e2e/client.test.ts [hunk 3]
```diff
@@ -1,8 +1,10 @@
-import { GenerateResult } from '../../api-types.js';
+import { GenerateOutput } from '../../api-types.js';
 import { GenerateInput } from '../../client-types.js';
 import { Client } from '../../client.js';
 import { RequestCanceledError } from '../../errors.js';
 
+export type GenerateResult = GenerateOutput['results'][0];
```
Comment (pilartomas) [inline src/tests/e2e/client.test.ts:6]: Do not reference API types in client tests (unless mocking)

File: src/api-types.ts [hunk 4]
```diff
@@ -338,3 +298,27 @@ export const PromptTemplateExecuteOutputSchema = z.object({
 export type PromptTemplateExecuteOutput = z.infer<
   typeof PromptTemplateExecuteOutputSchema
 >;
+
+// History
+export const HistoryInputSchema = z
+  .object({
+    status: z.enum(['SUCCESS', 'ERROR']),
+    origin: z.enum(['API', 'UI']),
```
Comment (pilartomas) [inline src/api-types.ts:306]: Please export these schemas separately.

File: src/api-types.ts [hunk 5]
```diff
@@ -60,29 +60,30 @@ export const GenerateInputSchema = z.object({
...
 
-export type GenerateStopReason =
-  | 'NOT_FINISHED'
-  | 'MAX_TOKENS'
-  | 'EOS_TOKEN'
-  | 'CANCELLED'
-  | 'TIME_LIMIT'
-  | 'STOP_SEQUENCE'
-  | 'TOKEN_LIMIT'
-  | 'ERROR';
```
Comment (pilartomas) [inline src/api-types.ts:71]: Keep the reason exported separately (and add schema export).

File: src/api-types.ts [hunk 6]
```diff
@@ -60,29 +60,30 @@ export const GenerateInputSchema = z.object({
...
-
-export interface GenerateResult {
```
Comment (pilartomas) [inline src/api-types.ts:73]: Backwards incompatible but good riddance. Same for the `Request*` types.

**Review by pilartomas — CHANGES_REQUESTED** [review]

File: src/api-types.ts (hunk 6 above)
Comment (Tomas2D) [reply src/api-types.ts:73]: This should not be a problem as API types are internal.

Comment (pilartomas) [reply src/api-types.ts:73]: No they're not, we are exporting them out of the package (on purpose).

File: src/client-types.ts [hunk 7]
```diff
@@ -165,6 +165,6 @@ export type HistoryInput = z.input<typeof HistoryInputSchema>;
 export type HistoryOptions = HttpHandlerOptions;
 
 export const HistoryOutputSchema = ApiTypes.HistoryOutputSchema;
-export type HistoryOutput = z.output<
-  typeof HistoryOutputSchema
->['results'][number];
+export type HistoryOutput = z.infer<
+  typeof HistoryOutputSchema.shape.results.element
+>;
```
Comment (pilartomas) [inline src/client-types.ts]: the `.shape.results.element` is meant for the `HistoryOutputSchema`

**Review by pilartomas — APPROVED** [review]

**Pull request merged**